odoo/addons/base/tests/test_ir_mail_server.py

```diff
@@ -336,7 +337,7 @@ def test_mail_server_binary_arguments_domain(self):
         self.assertFalse(self.env['ir.mail_server'].search([]))
 
         # Use an email in the domain of the "from_filter"
-        with self.mock_smtplib_connection():
+        with self.mock_smtplib_connection(config_smtp_server='localhost'):
```

Comment:
why is this required ? (it's weird that it try to read the odoo-bin SMTP config, because you created mail server in `_init_mail_servers`)

(same for all other example below)

Reply:
Because here you test when not having any mail servers. If I don't set the smtp, all sub tests crash (no mail servers, no from, cracboum). I just update tests you wrote so maybe that's not what was intended to be tested ?

Reply:
ah yes indeed, can't you do something like this ?
```python
@patch.dict("odoo.tools.config.options", {"from_filter": "test.com", "smtp_server": "example.com"})
```

Reply:
Indeed :)